packages/web3-react/src/hooks/useWallet.tsx

```diff
@@ -19,23 +19,49 @@ import { useMemo } from 'react'
 import { useAlephiumConnectContext } from '../contexts/alephiumConnect'
 import { NodeProvider, SignerProvider, Account, NetworkId } from '@alephium/web3'
 
-export interface Wallet {
-  signer: SignerProvider
-  account: Account & { network: NetworkId }
-  nodeProvider?: NodeProvider
-}
+export type Wallet =
+  | {
+      signer: SignerProvider
+      account: Account & { network: NetworkId }
+      connectionStatus: 'connected'
+      nodeProvider: NodeProvider | undefined
+    }
```

Comment:
Recommend placing `connectionStatus` at the first field so new devs knows that it is the type field.
```
    connectionStatus: 'connected';
    signer: SignerProvider;
    account: Account & {
        network: NetworkId;
    };
    nodeProvider: NodeProvider | undefined;
```